kbfsmd/root_metadata_v3.go

```diff
@@ -645,11 +645,35 @@ func (md *RootMetadataV3) CheckValidSuccessorForServer(
 	return nil
 }
 
+// IsBackedByTeam returns true if md is for a TLF backed by a team. It could be
+// either a SingleTeam TLF or a private/public TLF backed by an implicit team.
+func (md *RootMetadataV3) IsBackedByTeam(extra ExtraMetadata) bool {
+	if extra != nil {
+		return false
+	}
+	if len(md.WriterMetadata.UnresolvedWriters) != 0 {
+		return false
+	}
+	if len(md.WriterMetadata.Writers) != 1 {
+		return false
+	}
+	if _, err := md.WriterMetadata.Writers[0].AsTeam(); err != nil {
+		return false
+	}
+	return true
+}
+
 // MakeBareTlfHandle implements the RootMetadata interface for RootMetadataV3.
 func (md *RootMetadataV3) MakeBareTlfHandle(extra ExtraMetadata) (
 	tlf.Handle, error) {
+	if md.IsBackedByTeam(extra) {
+		return tlf.MakeHandle(md.WriterMetadata.Writers,
```

Comment:
How does this reconcile with the `TlfHandle` work that @strib is doing? Specifically, it seems that the handle produced on the client will contain the correct set of `{resolved,unresolved}{Readers,Writers}`, but the code here just seems to take the implicit team ID from the `md.WriterMetadata.Writers`.

Am I missing something?

Reply:
For team backed TLFs, bare handles would be mostly empty. It only contains an (implicit) team ID in the resolved writers list. In fact we use this attribute to determine whether it's a team backed TLF (i.e. SingleTeam or iTeam-backed TLF) or not.